app/src/main/java/org/simple/clinic/widgets/qrcodescanner/QrCodeScannerView.kt

```diff
@@ -13,26 +14,18 @@ import com.budiyev.android.codescanner.ScanMode
 import com.google.zxing.BarcodeFormat
 import com.jakewharton.rxbinding2.view.RxView
 import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
 import org.simple.clinic.activity.TheActivity
-import org.simple.clinic.bindUiToController
-import org.simple.clinic.widgets.ScreenCreated
 import org.simple.clinic.widgets.ScreenDestroyed
 import org.simple.clinic.widgets.TheActivityLifecycle
-import org.simple.clinic.widgets.UiEvent
 import timber.log.Timber
 import javax.inject.Inject
 
 class QrCodeScannerView(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
```

Comment:
Can we extend `CodeScannerView` instead of using a FrameLayout and then adding this view?

Reply:
We could. I'll do this.

Reply:
Never mind, I tried doing it and the library has made `CodeScannerView` a `final` class, so we can't inherit from it.